src/array.cr

```diff
@@ -1066,15 +1066,17 @@ class Array(T)
   # Returns a new Array that is a one-dimensional flattening of self (recursively).
   #
   # That is, for every element that is an array, extract its elements into the new array.
+  # It is also possible to give it an Integer argument telling it how many Array levels to flatten.
   #
   # ```
   # s = [1, 2, 3]         # => [1, 2, 3]
   # t = [4, 5, 6, [7, 8]] # => [4, 5, 6, [7, 8]]
   # a = [s, t, 9, 10]     # => [[1, 2, 3], [4, 5, 6, [7, 8]], 9, 10]
   # a.flatten             # => [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+  # a.flatten(1)          # => [1, 2, 3, 4, 5, 6, [7, 8], 9, 10]
   # ```
-  def flatten
-    FlattenHelper(typeof(FlattenHelper.element_type(self))).flatten(self)
+  def flatten(steps : Int = -2)
+    FlattenHelper(typeof(FlattenHelper.element_type(self, steps))).flatten(self, steps)
```

Comment:
I am guessing might want to trap for `steps <= -1` here [probably better performance]; otherwise, you'll probably want to do something like `if steps <= -1` checks in the loops below in FlattenHelper [probably a bit slower]; otherwise FlattenHelper's loops could continue towards -infinity [well probably -max int].